src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/data/scaler_methods/max_abs_scaler.hpp>
+#include <mlpack/core/data/scaler_methods/mean_normalization.hpp>
+#include <mlpack/core/data/scaler_methods/min_max_scaler.hpp>
+#include <mlpack/core/data/scaler_methods/whitening.hpp>
+#include <mlpack/core/data/scaler_methods/standard_scaler.hpp>
+
+PROGRAM_INFO("Scale Data",
+    // Short description.
+    "A utility to perform feature scaling.",
+    // Long description.
+    "This utility takes a dataset and performs feature scaling using one of "
+    "the five scaler methods namely: max_abs_scaler, mean_normalization, "
+    "min_max_scaler ,standard_scaler and whitening. The function takes a "
+    "matrice as " + PRINT_PARAM_STRING("input") + " and a scaling method type"
+    " which you can specify using " + PRINT_PARAM_STRING("scaler_method") +
+    " parameter; the default is standard scaler, and outputs a matrice "
+    "with scaled feature."
+    "\n\n"
+    "The output scaled feature matrices may be saved with the " +
+    PRINT_PARAM_STRING("output") + " output parameters."
+    "\n\n"
+    "So, a simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled")+ " with "
+    " standard_scaler as scaler_method, we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "standard_scaler") +
+    "\n\n"
+    "A simple example where we want to whiten the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_whitened")+ " with "
+    " PCA as whitening_method and use 0.01 as regularization parameter, "
+    "we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "whitening", "whitening_method", "PCA", "epsilon", 0.01) +
+    "\n\n"
+    "Another simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled") + " with "
+    " min_max_scaler as scaler method, where scaling range is 1 to 3 instead"
+    " of deafult 0 to 1. We coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "min_max_scaler", "min_value", 1, "max_value", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save scaled data to.", "o");
+PARAM_STRING_IN("scaler_method", "method to use for scaling, the "
+    "default is standard_scaler.", "a", "standard_scaler");
+PARAM_STRING_IN("whitening_method", "method to use for whitening, the "
+    "default is ZCA.", "w", "ZCA");
+PARAM_DOUBLE_IN("epsilon", "regularization Parameter, should be between"
+  "-1 to 1.", "r", 0.000001);
```

Comment:
This should be doubly indented since it's a wrapped line.

Reply:
Agreed, Made change.